### ISSUE TITLE
[Aula 06 - Condicionais: Apoio para quem tem dúvida]

### ISSUE THREAD

**Issue by SchmidtCarlos** [cwi-reset/edicao-04-level-1#314]
Buenas, pessoal!!! Enquanto estudava me surgiu a dúvida da diferença entre o uso do "If...else..." e o "Switch...case". Para tanto, fui pesquisar um pouco sobre essas diferenças e acredito ter entendido um pouco melhor. Por isso, trago aqui um pequeno apoio conforme o que eu entendi. (Caso esteja errado meu entendimento, por favor, podem me corrigir!!!). (Referência utilizada: https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Statements/switch)

Vamos lá:

O "if...else" irá utilizar o resultado obtido na variável "let" para tomar a decisão necessária. Como no exemplo abaixo, será utilizado "duvida1" para que tome tal decisão. Se a let "ajuda" receber o valor "duvida1", será mostrado "Temos a mesma dúvida", do contrário, mostrará "Que bom que você aprendeu, continue assim". Vejamos:

```javascript

let ajuda = "duvida1";

if (ajuda == "duvida1") {
  console.log("Temos a mesma dúvida");
} else {
  console.log("Que bom que você aprendeu, continue assim");
}
```

Para o uso do "switch...case", será tomado como base o que for definido como let, neste caso "ajuda", para que assim seja atribuída uma resposta e executado o código necessário para tal. Ou seja, não é necessário que tenha um valor atribuído para a let logo de início, fazendo com que seja esperado um valor para que assim seja fornecida a resposta. Seguimos o exemplo:

```javascript

let ajuda = "duvida1";

switch (ajuda) {
   case "duvida1":
      console.log("Temos a mesma dúvida");
     break;
   default:
      console.log("Que bom que você aprendeu, continue assim");
}
```

Espero que tenha ajudado a todos! 
Grande abraço! #ResetCWI

**Comment by JefersonPerchim** [comment]
Sim, Swicth...Case... compara os valores que estão na variável com aqueles valores pré-definidos, embora seja um pouco limitado é bastante útil e prático. Já o if...else.., vc consegue elaborar uma equação logica, utilizando os operadores lógicos, o que abrange um campo maior de possibilidades.
No final, nenhuma das duas funções é melhor que a outra, cada uma com suas particularidades.
Eu acho bem legal quando surge uma possibilidade de utilizar as duas.

**Comment by marinanervo** [comment]
Show @SchmidtCarlos, obrigada por compartilhar :)

**Comment by michelefagundes** [comment]
Bah, Carlos, ajudou demais! Tenho um pouco de dificuldade nessa parte. 

Valeuzão
